File: main/insertion-sort-list/insertion-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def insertionSortList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dest_sentinel = ListNode(None)

        while head is not None:
            dest = dest_sentinel
            while dest.next is not None and dest.next.val < head.val:
                dest = dest.next

            next_head = head.next
            head.next = dest.next
            dest.next = head

            head = next_head

        return dest_sentinel.next

File: main/insertion-sort-list/insertion-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode insertionSortList(ListNode head) {
        final ListNode destSentinel = new ListNode(0);

        while (head != null) {
            ListNode dest = destSentinel;
            while (dest.next != null && dest.next.val < head.val)
                dest = dest.next;

            final ListNode next = head.next;
            head.next = dest.next;
            dest.next = head;

            head = next;
        }

        return destSentinel.next;
    }
}
